shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -444,11 +444,14 @@ bool PlatformView::OnHandlePointerEvent(
   pointer_data.time_stamp = pointer.event_time / 1000;
   pointer_data.change = GetChangeFromPointerEventPhase(pointer.phase);
   pointer_data.kind = GetKindFromPointerType(pointer.type);
-  pointer_data.device = pointer.pointer_id;
+  pointer_data.device = pointer.device_id;
   pointer_data.physical_x = pointer.x * metrics_.scale;
   pointer_data.physical_y = pointer.y * metrics_.scale;
   // Buttons are single bit values starting with kMousePrimaryButton = 1.
   pointer_data.buttons = static_cast<uint64_t>(pointer.buttons);
+  // This is currently overriden by platform_data_packet_converter, but
+  // leaving this here incase we wish to utilize this.
+  pointer_data.pointer_identifier = pointer.pointer_id;
 
   switch (pointer_data.change) {
     case flutter::PointerData::Change::kDown:
```

Comment:
This should be  down_pointers_.insert(pointer_data. pointer_identifier);
and
down_pointers_.erase(pointer_data. pointer_identifier);

I think in the contest of this file it does not really matter as long as it is unique identifier for the pointer, but using pointer_id is more clear.

device_id: is the unique id for the device
pointer_id: is the unique id for each pointer down. A device might use multiple pointer id in each different pointer down event.
For example, i have a mouse device id is 0
when i first click the screen I will be assigned a pointer_id= 10, so i will send pointerdown(device =0, pointer_id=10)
after that i release the click, i will send pointerup(device =0, pointer_id=10)

10 mins later, i click the mouse again. I will be assigned a new pointer_id = 143, so I will send pointerdown(device =0, pointer_id=143)
... and so on

Reply:
Good call, I will close this PR, add some tests and then re-submit.